tests/test_linalg.py

```diff
@@ -508,3 +508,82 @@ def test_svd(type, shape):
         assert_allclose(
             inp.asnumpy(dpnp_vt)[i, :], np_vt[i, :], rtol=tol, atol=tol
         )
+
+
+class TestSolve:
+    @pytest.mark.parametrize("dtype", get_all_dtypes(no_bool=True))
+    def test_solve(self, dtype):
+        a_np = numpy.array([[1, 0.5], [0.5, 1]], dtype=dtype)
+        a_dp = inp.array(a_np)
+
+        expected = numpy.linalg.solve(a_np, a_np)
+        result = inp.linalg.solve(a_dp, a_dp)
+
+        assert_allclose(expected, result, rtol=1e-06)
+
+    @pytest.mark.parametrize("a_dtype", get_all_dtypes(no_bool=True))
+    @pytest.mark.parametrize("b_dtype", get_all_dtypes(no_bool=True))
+    def test_solve_diff_type(self, a_dtype, b_dtype):
+        a_np = numpy.array([[1, 2], [3, -5]], dtype=a_dtype)
+        b_np = numpy.array([4, 1], dtype=b_dtype)
+
+        a_dp = inp.array(a_np)
+        b_dp = inp.array(b_np)
+
+        expected = numpy.linalg.solve(a_np, b_np)
+        result = inp.linalg.solve(a_dp, b_dp)
+
+        assert_allclose(expected, result, rtol=1e-06)
+
+        if has_support_aspect64():
+            assert expected.dtype == result.dtype
+        else:
+            assert expected.dtype.kind == result.dtype.kind
```

Comment:
is not the same like `assert_dtype_allclose` does?

Reply:
Yes, you are right
I rewrote that part of the code using `assert_dtype_allclose`.
Thanks